cwms-data-api/src/main/java/cwms/cda/api/location/kind/GateChangeCreateController.java

```diff
@@ -0,0 +1,83 @@
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package cwms.cda.api.location.kind;
+
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer;
+import cwms.cda.api.BaseHandler;
+import cwms.cda.data.dao.JooqDao;
+import cwms.cda.data.dao.location.kind.OutletDao;
+import cwms.cda.data.dto.location.kind.GateChange;
+import cwms.cda.formatters.ContentType;
+import cwms.cda.formatters.Formats;
+import io.javalin.core.util.Header;
+import io.javalin.http.Context;
+import io.javalin.plugin.openapi.annotations.HttpMethod;
+import io.javalin.plugin.openapi.annotations.OpenApi;
+import io.javalin.plugin.openapi.annotations.OpenApiContent;
+import io.javalin.plugin.openapi.annotations.OpenApiParam;
+import io.javalin.plugin.openapi.annotations.OpenApiRequestBody;
+import io.javalin.plugin.openapi.annotations.OpenApiResponse;
+import java.util.List;
+import javax.servlet.http.HttpServletResponse;
+import org.jetbrains.annotations.NotNull;
+import org.jooq.DSLContext;
+import static cwms.cda.api.Controllers.*;
+
+public class GateChangeCreateController extends BaseHandler {
+
+    public GateChangeCreateController(MetricRegistry metrics) {
+        super(metrics);
+    }
+
+    @OpenApi(
+            requestBody = @OpenApiRequestBody(
+                    content = {
+                            @OpenApiContent(from = GateChange.class, isArray = true, type = Formats.JSONV1),
+                            @OpenApiContent(from = GateChange.class, isArray = true, type = Formats.JSON)
+                    },
+                    required = true),
+            queryParams = {
+                    @OpenApiParam(name = FAIL_IF_EXISTS, type = Boolean.class,
+                            description = "Create will fail if provided Gate Changes already exist. Default: true")
+            },
+            description = "Create CWMS Gate Changes",
+            method = HttpMethod.POST,
+            tags = {OutletController.TAG},
+            responses = {
+                    @OpenApiResponse(status = STATUS_201, description = "Gate Changes successfully stored to CWMS.")
+            }
+    )
+    @Override
+    public void handle(@NotNull Context context) throws Exception {
+        boolean failIfExists = context.queryParamAsClass(FAIL_IF_EXISTS, Boolean.class).getOrDefault(true);
+        String formatHeader = context.header(Header.ACCEPT) != null ? context.header(Header.ACCEPT) : Formats.JSONV1;
+        ContentType contentType = Formats.parseHeader(formatHeader, GateChange.class);
```

Comment:
Should we just have parseHeader return the configured default if null is passed in, or an error if there is not default?

Reply:
NOTE: we don't need to do that in this PR, just realize how common this could be and we are trying to go for convention over configuration here... well or at least put the onerous on format choices on the DTOs not the controllers anyways.

Reply:
Yeah, I'm not saying we change it now, I just keep seeing this type of two line code repeated and am wondering if we should shift the responsibility in a future PR.

Reply:
We already have a DEFAULT constant that's handled in the annotations (or at least can be), I agree that we should just have the `parseHeader` substitute null for that DEFAULT and then controllers no longer need to null check. I also agree that a separate PR makes sense for that change. @RyanM-RMA - I can do that change if you're in agreement

Reply:
probably better at that level to return "*/*" which would make sure the system just went through and found the right default for the DTO.

Reply:
Yeah, I can't think of why we would have any nulls that wouldn't be handled the same as `DEFAULT` (which is `*/*`)

Reply:
Oh I see what you mean @RyanM-RMA - if there is no matching content type for the DTO and the we pass in `*/*` it gets replaced with "application/json".  I think that should still be ok.

Reply:
I think we should resolve it to the appropriate version for the class, given the aliasing.

Reply:
If the user passes in null, but the DTO has no `DEFAULT` handling, how do we know what is the appropriate version? The first annotation?

Reply:
Error, we should assume if no default was set, that it was intentional.

Reply:
Handling under: https://github.com/USACE/cwms-data-api/pull/861